core/solver/multigrid.cpp

```diff
@@ -554,17 +568,26 @@ void Multigrid::generate()
             using value_type =
                 typename std::decay_t<decltype(*mg_level)>::value_type;
             auto exec = this->get_executor();
+            // default coarse grid solver, direct LU
+            // TODO: maybe remove fixed index type
+            auto gen_default_solver = [&] {
+                return experimental::solver::Direct<value_type, int32>::build()
```

Comment:
Could we use the experimental class in another public class?

Reply:
I don't see why not. The default initialization is internal anyway.

Reply:
The experimental features can't be turned off. Only the experimental MPI stuff is disabled if MPI is not available.